bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/internal/framework/BundleContextImpl.java

```diff
@@ -846,8 +846,18 @@ Object setContextFinder() {
 		ClassLoader previousTCCL = currentThread.getContextClassLoader();
 		ClassLoader contextFinder = container.getContextFinder();
 		if (previousTCCL != contextFinder) {
-			currentThread.setContextClassLoader(container.getContextFinder());
-			return previousTCCL;
+			try {
+				currentThread.setContextClassLoader(container.getContextFinder());
+				return previousTCCL;
+			} catch (SecurityException e) {
```

Comment:
I think we should catch `RuntimeException` here.  Given that the security manager deprecation/removal I look forward to the Java team wanting to throw some kind of `RuntimeException` here in the future instead.  I think that would be more safe, besides that, any reason for failing to set the TCCL here should not prevent the rest of the code flow.